specs/sharding/beacon-chain.md

```diff
@@ -103,786 +87,558 @@ The following values are (non-configurable) constants used throughout the specif
+    """
+    Verifies that the commitment is of polynomial degree <= degree. 
+    """
+
+    assert (
+        bls.Pairing(proof, G2_SETUP[0])
+        == bls.Pairing(commitment, G2_SETUP[-degree - 1])
+    )
 ```
 
-#### `compute_committee_source_epoch`
+#### `block_to_field_elements`
 
 ```python
-def compute_committee_source_epoch(epoch: Epoch, period: uint64) -> Epoch:
+def block_to_field_elements(block: bytes) -> List[BLSFieldElement]:
     """
-    Return the source epoch for computing the committee.
+    Slices a block into 31 byte chunks that can fit into field elements
     """
-    source_epoch = Epoch(epoch - epoch % period)
-    if source_epoch >= period:
-        source_epoch -= period  # `period` epochs lookahead
-    return source_epoch
+    sliced_block = [block[i:i + 31] for i in range(0, len(bytes), 31)]
+    return [BLSFieldElement(int.from_bytes(x, "little")) for x in sliced_block]
 ```
 
-#### `batch_apply_participation_flag`
+#### `roots_of_unity`
 
 ```python
-def batch_apply_participation_flag(state: BeaconState, bits: Bitlist[MAX_VALIDATORS_PER_COMMITTEE],
-                                   epoch: Epoch, full_committee: Sequence[ValidatorIndex], flag_index: int):
-    if epoch == get_current_epoch(state):
-        epoch_participation = state.current_epoch_participation
-    else:
-        epoch_participation = state.previous_epoch_participation
-    for bit, index in zip(bits, full_committee):
-        if bit:
-            epoch_participation[index] = add_flag(epoch_participation[index], flag_index)
-```
+def roots_of_unity(order: uint64) -> List[BLSFieldElement]:
+    r = []
+    root_of_unity = pow(PRIMITIVE_ROOT_OF_UNITY, (MODULUS - 1) // order, MODULUS)
 
-### Beacon state accessors
+    current_root_of_unity = 1
+    for i in range(len(SAMPLES_PER_BLOB * FIELD_ELEMENTS_PER_SAMPLE)):
+        r.append(current_root_of_unity)
+        current_root_of_unity = current_root_of_unity * root_of_unity % MODULUS
+    return r
+```
 
-#### Updated `get_committee_count_per_slot`
+#### `modular_inverse`
 
 ```python
-def get_committee_count_per_slot(state: BeaconState, epoch: Epoch) -> uint64:
-    """
-    Return the number of committees in each slot for the given ``epoch``.
-    """
-    return max(uint64(1), min(
-        get_active_shard_count(state, epoch),
-        uint64(len(get_active_validator_indices(state, epoch))) // SLOTS_PER_EPOCH // TARGET_COMMITTEE_SIZE,
-    ))
+def modular_inverse(a):
+    assert(a == 0):
+    lm, hm = 1, 0
+    low, high = a % MODULUS, MODULUS
+    while low > 1:
+        r = high // low
+        nm, new = hm - lm * r, high - low * r
+        lm, low, hm, high = nm, new, lm, low
+    return lm % MODULUS
 ```
 
-#### `get_active_shard_count`
+#### `eval_poly_at`
 
 ```python
-def get_active_shard_count(state: BeaconState, epoch: Epoch) -> uint64:
+def eval_poly_at(poly: List[BLSFieldElement], x: BLSFieldElement) -> BLSFieldElement:
```

Comment:
I would recommend changing the function name to reflect that this is taking as input evaluation form and not coeffs.